Surface3d example is broken

The extension currently fails to compile cc @mattpap 
```
dev312 ❯ bokeh serve --show surface3d
2025-02-08 17:12:40,641 Starting Bokeh server version 3.7.0.dev5+28.g65e644fd (running on Tornado 6.4.2)
2025-02-08 17:12:40,642 User authentication hooks NOT provided (default user enabled)
2025-02-08 17:12:40,643 Bokeh app running at: http://localhost:5006/surface3d
2025-02-08 17:12:40,643 Starting Bokeh server with process id: 42504
Compilation failed:

surface3d/surface3d.ts:86:12 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Data'.
  No index signature with a parameter of type 'string' was found on type 'Data'.

86         x: source.data[this.model.x][i],
              ~~~~~~~~~~~~~~~~~~~~~~~~~
surface3d/surface3d.ts:87:12 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Data'.
  No index signature with a parameter of type 'string' was found on type 'Data'.

87         y: source.data[this.model.y][i],
              ~~~~~~~~~~~~~~~~~~~~~~~~~
surface3d/surface3d.ts:88:12 - error TS7053: Element implicitly has an 'any' type because expression of type 'string' can't be used to index type 'Data'.
  No index signature with a parameter of type 'string' was found on type 'Data'.

88         z: source.data[this.model.z][i],
              ~~~~~~~~~~~~~~~~~~~~~~~~~
surface3d/surface3d.ts:138:34 - error TS2345: Argument of type '({ Unknown, Str, Dict, Ref }: typeof import("/Users/bryan/work/bokeh/src/bokeh/server/static/js/lib/core/kinds")) => { x: [Str]; y: [Str]; z: [Str]; data_source: [Ref<ColumnDataSource>]; options: [Dict<unknown>, { width: string; height: string; style: string; showPerspective: boolean; showGrid: boolean; keepAspectRatio: boolean; verticalRatio: number; legendLabel: string; cameraPosition: { horizontal: number; vertical: number; distance: number; }; }]; }' is not assignable to parameter of type 'Partial<DefineOf<Props, HasProps>> | ((types: typeof import("/Users/bryan/work/bokeh/src/bokeh/server/static/js/lib/core/kinds")) => Partial<DefineOf<Props, HasProps>>)'.
  Type '({ Unknown, Str, Dict, Ref }: typeof import("/Users/bryan/work/bokeh/src/bokeh/server/static/js/lib/core/kinds")) => { x: [Str]; y: [Str]; z: [Str]; data_source: [Ref<ColumnDataSource>]; options: [Dict<unknown>, { width: string; height: string; style: string; showPerspective: boolean; showGrid: boolean; keepAspectRatio: boolean; verticalRatio: number; legendLabel: string; cameraPosition: { horizontal: number; vertical: number; distance: number; }; }]; }' is not assignable to type '(types: typeof import("/Users/bryan/work/bokeh/src/bokeh/server/static/js/lib/core/kinds")) => Partial<DefineOf<Props, HasProps>>'.
    Call signature return types '{ x: [Str]; y: [Str]; z: [Str]; data_source: [Ref<ColumnDataSource>]; options: [Dict<unknown>, { width: string; height: string; style: string; showPerspective: boolean; showGrid: boolean; keepAspectRatio: boolean; verticalRatio: number; legendLabel: string; cameraPosition: { horizontal: number; vertical: number; distance: number; }; }]; }' and 'Partial<DefineOf<Props, HasProps>>' are incompatible.
      Type at position 0 in source is not compatible with type at position 0 in target.
        The types of 'options' are incompatible between these types.
          Type 'Dict<unknown>' is not assignable to type 'PropertyAlias | PropertyConstructor<{ [key: string]: unknown; }> | Kind<{ [key: string]: unknown; }>'.
            Type 'Dict<unknown>' is not assignable to type 'Kind<{ [key: string]: unknown; }>'.
              Types of property '__type__' are incompatible.
                Type 'Dict<unknown>' is not assignable to type '{ [key: string]: unknown; }'.
                  Type 'Map<string, unknown>' is not assignable to type '{ [key: string]: unknown; }'.
                    Index signature for type 'string' is missing in type 'Map<string, unknown>'.

138     this.define<Surface3d.Props>(({Unknown, Str, Dict, Ref}) => ({
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
139       x:           [ Str ],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
143       options:     [ Dict(Unknown), OPTIONS ],
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
144     }))
    ~~~~~~
```

--- a/examples/server/app/surface3d/surface3d.ts
+++ b/examples/server/app/surface3d/surface3d.ts
@@ -12,17 +12,17 @@
 import {LayoutDOM, LayoutDOMView} from "models/layouts/layout_dom"
 import {ColumnDataSource} from "models/sources/column_data_source"
 import {to_object} from "core/util/object"
-import type {Dict} from "core/types"
+import type {Dict, PlainObject} from "core/types"
 import * as p from "core/properties"
 
 declare namespace vis {
   class Graph3d {
-    constructor(el: HTMLElement | DocumentFragment, data: object, OPTIONS: object)
+    constructor(el: HTMLElement | DocumentFragment, data: DataSet, OPTIONS: PlainObject)
     setData(data: vis.DataSet): void
   }
 
   class DataSet {
-    add(data: unknown): void
+    add(data: PlainObject): void
   }
 }
 
@@ -67,7 +67,7 @@ export class Surface3dView extends LayoutDOMView {
     // Bokeh views ignore this default <div>, and instead do things like draw
     // to the HTML canvas. In this case though, we use the <div> to attach a
     // Graph3d to the DOM.
-    this._graph = new vis.Graph3d(this.shadow_el, this.get_data(), this.model.options)
+    this._graph = new vis.Graph3d(this.shadow_el, this.get_data(), to_object(this.model.options))
   }
 
   override connect_signals(): void {
@@ -142,7 +142,7 @@ export class Surface3d extends LayoutDOM {
       y:           [ Str ],
       z:           [ Str ],
       data_source: [ Ref(ColumnDataSource) ],
-      options:     [ Dict(Unknown), to_object(OPTIONS) ],
+      options:     [ Dict(Unknown), OPTIONS ],
     }))
   }
 }

--- a/examples/server/app/surface3d/surface3d.ts
+++ b/examples/server/app/surface3d/surface3d.ts
@@ -11,6 +11,8 @@
 
 import {LayoutDOM, LayoutDOMView} from "models/layouts/layout_dom"
 import {ColumnDataSource} from "models/sources/column_data_source"
+import {to_object} from "core/util/object"
+import type {Dict} from "core/types"
 import * as p from "core/properties"
 
 declare namespace vis {
@@ -83,9 +85,9 @@ export class Surface3dView extends LayoutDOMView {
     const source = this.model.data_source
     for (let i = 0; i < source.get_length()!; i++) {
       data.add({
-        x: source.data[this.model.x][i],
-        y: source.data[this.model.y][i],
-        z: source.data[this.model.z][i],
+        x: source.get(this.model.x)[i],
+        y: source.get(this.model.y)[i],
+        z: source.get(this.model.z)[i],
       })
     }
     return data
@@ -105,7 +107,7 @@ export namespace Surface3d {
     y: p.Property<string>
     z: p.Property<string>
     data_source: p.Property<ColumnDataSource>
-    options: p.Property<{[key: string]: unknown}>
+    options: p.Property<Dict<unknown>>
   }
 }
 
@@ -140,7 +142,7 @@ export class Surface3d extends LayoutDOM {
       y:           [ Str ],
       z:           [ Str ],
       data_source: [ Ref(ColumnDataSource) ],
-      options:     [ Dict(Unknown), OPTIONS ],
+      options:     [ Dict(Unknown), to_object(OPTIONS) ],
     }))
   }
 }

